### PR TITLE
Week 10 - Animation in Jetpack Compose

### PR DESCRIPTION
**Pull request by ibrohimkhan**
New features:

- _Shimmer_ animation option for _Loading_ composable
- _Splash_ screen with _Lottie_ animation
- Star icon animation in country list
- Flag animation in country details screen
- Animation in Screen navigation

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -25,15 +25,13 @@ android {
     }
 
     buildTypes {
-        debug {
-            isMinifyEnabled = false
-        }
         release {
-            isMinifyEnabled = true
+            isMinifyEnabled = false
```
Comment (zlippard) [inline app/build.gradle.kts]: Likely this should be reverted as you want minification enabled for the `release` build type.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -25,15 +25,13 @@ android {
...
             proguardFiles(
                 getDefaultProguardFile("proguard-android-optimize.txt"),
                 "proguard-rules.pro"
             )
+            signingConfig = signingConfigs.getByName("debug")
```
Comment (zlippard) [inline app/build.gradle.kts]: Was this debug signing config needed?

File: app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt [hunk 3]
```diff
@@ -32,7 +33,17 @@ class MainActivity : ComponentActivity() {
...
+
+                    LaunchedEffect(key1 = Unit) {
+                        delay(3000)
+                        isAnimationFinished = true
+                    }
+
+                    when (isAnimationFinished) {
+                        false -> SplashScreen()
+                        true -> ApplicationNavigation(repository)
+                    }
```
Comment (zlippard) [inline app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt]: Nice job here! As mentioned you could also do this directly in the ApplicationNavigation where the navhost lives.

```kotlin
            composable(route = Screens.Splash.path) {
                LaunchedEffect(null) {
                    delay(3000)
                    navController.navigate(Screen.CountryList.route){
                        popUpTo(Screen.Splash.route){
                            inclusive = true
                        }
                        launchSingleTop = true
                    }
                }
                SplashScreen()
```

File: app/src/main/java/com/kodeco/android/countryinfo/utils/Extensions.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.compose.animation.core.rememberInfiniteTransition
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+
+fun Modifier.shimmerLoadingAnimation(
```
Comment (zlippard) [inline app/src/main/java/com/kodeco/android/countryinfo/utils/Extensions.kt:16]: Great job on this shimmer modifier!! 👏

**Review by zlippard — APPROVED** [review]
Awesome job this week @ibrohimkhan on the animations! Also amazing job going above and beyond with a splash screen lottie animation and the shimmer modifier as well.  I left a few comments with some suggestions but overall things are looking great! Keep it up! :]

File: app/build.gradle.kts (hunk 1 above)
Comment (ibrohimkhan) [reply app/build.gradle.kts]: this is just for experimentation

File: app/build.gradle.kts (hunk 2 above)
Comment (ibrohimkhan) [reply app/build.gradle.kts]: this one is also for experimentation

Comment (ibrohimkhan) [reply app/build.gradle.kts]: Reverted it

File: app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt (hunk 3 above)
Comment (ibrohimkhan) [reply app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt]: thanks, this helped me a lot!

**Pull request merged**